### PR TITLE
Add --exclude so that you can exclude files/folders from being formatted

### PR DESCRIPTION
**Pull request by aliak00**

### CODE REVIEW

**Comment by nicklockwood** [comment]
Perfect, thanks!

File: Tests/SwiftFormatTests.swift [hunk 1]
```diff
@@ -82,6 +82,19 @@ class SwiftFormatTests: XCTestCase {
...
+        let currentFile = URL(fileURLWithPath: #file)
+        let excludedURLs = [currentFile.deletingLastPathComponent()]
+        let inputURL = currentFile.deletingLastPathComponent().deletingLastPathComponent()
+        let errors = enumerateFiles(withInputURL: inputURL, excluding: excludedURLs, outputURL: inputURL) { inputURL, outputURL in
+            XCTAssertEqual(inputURL, outputURL)
+            return { files.append(inputURL) }
+        }
+        XCTAssertEqual(errors.count, 0)
+        XCTAssertEqual(files.count, 15)
+    }
```
Comment (nicklockwood) [inline Tests/SwiftFormatTests.swift]: 😍

**Comment by nicklockwood** [comment]
I'll merge this now, but I've got a couple of other features to finish before I cut another release. This will be included in version 0.27, which should ship in a day or two. Thanks again!

**Comment by aliak00** [comment]
Awesome! And no worries 😄

**Pull request merged**